File: Server/src/main/java/com/api/server/foodb/CompoundOntologyTerm.java
package com.api.server.foodb;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "compound_ontology_terms")
public class CompoundOntologyTerm {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "compound_id")
    private Integer compoundId;

    @Column(name = "export")
    private Boolean export;

    @Column(name = "ontology_term_id")
    private Integer ontologyTermId;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/CompoundOntologyTerm.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DotNetServer.Core.Entities.Foodb;

[Table("compound_ontology_terms")]
public class CompoundOntologyTerm
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("compound_id")] public int? CompoundId { get; set; }

    [Column("export")] public bool? Export { get; set; }

    [Column("ontology_term_id")] public int? OntologyTermId { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }
}